### PR TITLE
Migrate xpallet-asset-registrar to pallet v2

### PR DESCRIPTION
**Pull request by VegeBun-csj**
image

### CODE REVIEW

**Comment by VegeBun-csj** [comment]
### pass all the tests
image

File: xpallets/assets-registrar/src/lib.rs [hunk 1]
```diff
@@ -85,27 +94,71 @@ decl_error! {
...
-        pub AssetInfoOf get(fn asset_info_of): map hasher(twox_64_concat) AssetId => Option<AssetInfo>;
-
-        /// The map of asset to the online state.
-        pub AssetOnline get(fn asset_online): map hasher(twox_64_concat) AssetId => bool;
-
-        /// The map of asset to the block number at which the asset was registered.
-        pub RegisteredAt get(fn registered_at): map hasher(twox_64_concat) AssetId => T::BlockNumber;
+    #[pallet::storage]
+    #[pallet::getter(fn asset_ids_of)]
+    /// Asset id list for each Chain.
```
Comment (liuchengxu) [inline xpallets/assets-registrar/src/lib.rs]:
```suggestion
    /// Asset id list for each Chain.
    #[pallet::storage]
    #[pallet::getter(fn asset_ids_of)]
```

**Review by liuchengxu — CHANGES_REQUESTED** [review]
Merge the latest develop branch and then make the ci pass.

**Review by hacpy — COMMENTED** [review]
It is recommended to rearrange the order of pallet components in the following order:
- pallet::config
- pallet::pallet
- pallet::call
- pallet::event
- pallet::error
- pallet::storage

**Pull request merged**